opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryMetricsExporter.java

```diff
@@ -0,0 +1,206 @@
+              Objects.requireNonNull(timeSeries.getLabelValues().get(i).getValue()));
+        }
+        Labels labels = labelsBuilder.build();
+        ArrayList<MetricData.Point> points = new ArrayList<>();
+        MetricDescriptor.Type type = null;
+        for (Point point : timeSeries.getPoints()) {
+          long timestampNanos =
+              TimeUnit.SECONDS.toNanos(point.getTimestamp().getSeconds())
+                  + point.getTimestamp().getNanos();
+          type = metric.getMetricDescriptor().getType();
+          switch (type) {
+            case GAUGE_INT64:
+            case CUMULATIVE_INT64:
+              points.add(
+                  LongPoint.create(
+                      timestampNanos,
+                      timestampNanos,
+                      labels,
+                      point
+                          .getValue()
+                          .match(
+                              Double::longValue,
+                              arg -> arg,
+                              arg -> null,
+                              arg -> null,
+                              arg -> null)));
+              break;
+            case GAUGE_DOUBLE:
+            case CUMULATIVE_DOUBLE:
+              points.add(
+                  DoublePoint.create(
+                      timestampNanos,
+                      timestampNanos,
+                      labels,
+                      point
+                          .getValue()
+                          .match(
+                              arg -> arg,
+                              Long::doubleValue,
+                              arg -> null,
+                              arg -> null,
+                              arg -> null)));
+              break;
+            case SUMMARY:
+              points.add(
+                  SummaryPoint.create(
+                      timestampNanos,
+                      timestampNanos,
+                      labels,
+                      point
+                          .getValue()
+                          .match(
+                              arg -> null,
+                              arg -> null,
+                              arg -> null,
+                              Summary::getCount,
+                              arg -> null),
+                      point
+                          .getValue()
+                          .match(
+                              arg -> null, arg -> null, arg -> null, Summary::getSum, arg -> null),
+                      point
+                          .getValue()
+                          .match(
+                              arg -> null,
+                              arg -> null,
+                              arg -> null,
+                              OpenTelemetryMetricsExporter::mapPercentiles,
+                              arg -> null)));
+              break;
+            default:
+              LOGGER.warning(type + " not supported by OpenCensus to OpenTelemetry migrator.");
```

Comment:
This message will be printed for every data point - maybe we should aggregate detected unsupported types and print one message per export?